Portfolio/src/main/java/com/google/sps/servlets/CreateCommentsServlet.java

```diff
@@ -15,18 +15,31 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
-public class DataServlet extends HttpServlet {
+import java.util.stream.Collectors;
+import com.google.gson.Gson;
+
+@WebServlet("/create_comment")
+public class CreateCommentsServlet extends HttpServlet {
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String body = request.getReader().lines().collect(Collectors.joining());
+    Map data = new Gson().fromJson(body, Map.class);
+    String username = data.get("username").toString();
+    String message = data.get("message").toString();
+
+    Comment comment = new Comment(username, message);
+    comment.toDataStore();
```

Comment:
Interesting, I like this idea of abstracting away database details from us here and leaving them up to the Comment class. However, my concerns come when the Database interaction could be expensive in nature, in which we'd actually want the database to be abstracted away from the model in this case.

Would it be a better design for `toDataStore` to return an `Entity` in which the caller could write to Datastore? Combining this with my coherence suggestion, you could make the `datastore` variable a private member of `CommentsServlet` and re-use it between doGet and doPost respectively.

Hopefully this makes sense!

Reply:
I agree with your idea to abstract away database details, good design idea